src/lib.rs

```diff
@@ -130,6 +132,27 @@ pub fn reformat_string(text: &str) -> String {
     }
 }
 
+#[derive(Debug)]
+pub enum ReformatFile {
+    Change,
+    NoChange,
+}
+
+pub fn reformat_file(
+    file: &PathBuf,
+    write_changes: bool,
+) -> Result<ReformatFile, Box<dyn std::error::Error>> {
+    let input = fs::read_to_string(file)?;
+    let output = reformat_string(&input);
+    if input != output {
+        if write_changes {
+            fs::write(file, &output)?;
+        }
+        return Ok(ReformatFile::Change);
+    }
+    Ok(ReformatFile::NoChange)
+}
```

Comment:
Ok I moved it back for now. Initially I wanted to implement a streaming version, so that I could early-exit when just looking for changes, but nixpkgs-fmt is not really structured that way at the moment.